File: frontend/src/store/api/hooks.ts
import { http } from "./http";

// Mirrors backend GlobalHookEntry (backend/app/agents/hooks_catalog.py),
// scanned from hooks/global/{hook_id}/HOOK.md.
export interface GlobalHookEntry {
  id: string;
  name: string;
  display_name: string;
  description: string;
  content: string;
  event: string;
  trigger: string;
  compatible_agents: string[];
  isBeta?: boolean;
  tags: string[];
}

// Frontend hook model (normalized from GlobalHookEntry for UI consumption)
export interface HookDef {
  id: string;
  name: string;
  description: string;
  event: "PreToolUse" | "PostToolUse" | "Stop" | "SessionStart" | "SessionEnd";
  trigger: string;
  compatible_agents: string[];
  tags: string[];
}

export interface HooksLibraryFilterOption {
  id: string;
  label: string;
  icon?: string;
}

export interface HooksLibraryResponse {
  hooks: GlobalHookEntry[];
  total_count: number;
  events: HooksLibraryFilterOption[];
}

export const hooksLibraryApi = {
  library: async (): Promise<HooksLibraryResponse> => {
    const { data } = await http.get<HooksLibraryResponse>("/api/hooks/library");
    return data;
  },
};

File: backend/app/api/hooks.py
"""Global Hooks catalog API.

This is the read-only, shared catalog of available behavioral hooks
shown in the Library UI. Sourced from ``hooks/global/{hook_id}/HOOK.md``
via ``app.agents.hooks_catalog.list_global_hooks()`` — a folder scan, same
discovery pattern as ``agents/prompts/{agent_id}/AGENT.md`` (see
``agents/loader.py``). Adding a hook is a folder-drop; nothing here
needs to change. Events/tags are derived from the scanned entries, not
maintained as a separate list.
"""

import logging
from dataclasses import asdict

from fastapi import APIRouter, Depends

from app.agents.hooks_catalog import list_global_hooks, _get_icon_for_event
from app.core.dependencies import get_current_user
from app.models.user import User

logger = logging.getLogger("app.api.hooks")

router = APIRouter(prefix="/api/hooks", tags=["hooks"])


@router.get("/library")
def get_hooks_library(
    current_user: User = Depends(get_current_user),
):
    """Get the full global Hooks catalog (mirrors GET /api/skills/library)."""
    entries = list_global_hooks()

    events = sorted({e.event for e in entries if e.event})

    return {
        "hooks": [asdict(e) for e in entries],
        "total_count": len(entries),
        "events": [{"id": "all", "label": "All"}]
        + [{"id": e, "label": e.replace("_", " "), "icon": _get_icon_for_event(e)} for e in events],
    }
